StreamProcessorTest#shouldCreateSnapshot is flaky

```
10:20:14.556Z] [ERROR] shouldCreateSnapshot(io.zeebe.engine.processor.StreamProcessorTest)  Time elapsed: 3.042 s  <<< FAILURE!
[2019-06-26T10:20:14.556Z] org.mockito.exceptions.verification.VerificationInOrderFailure: 
[2019-06-26T10:20:14.556Z] 
[2019-06-26T10:20:14.556Z] Verification in order failure
[2019-06-26T10:20:14.556Z] Wanted but not invoked:
[2019-06-26T10:20:14.556Z] stateSnapshotController.takeTempSnapshot();
[2019-06-26T10:20:14.556Z] -> at io.zeebe.engine.processor.StreamProcessorTest.shouldCreateSnapshot(StreamProcessorTest.java:516)
[2019-06-26T10:20:14.556Z] Wanted anywhere AFTER following interaction:
[2019-06-26T10:20:14.556Z] stateSnapshotController.openDb();
[2019-06-26T10:20:14.556Z] -> at io.zeebe.engine.util.TestStreams.buildStreamProcessor(TestStreams.java:266)
[2019-06-26T10:20:14.556Z] 
[2019-06-26T10:20:14.556Z] 	at io.zeebe.engine.processor.StreamProcessorTest.shouldCreateSnapshot(StreamProcessorTest.java:516)
[2019-06-26T10:20:14.556Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-06-26T10:20:14.556Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-06-26T10:20:14.556Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-06-26T10:20:14.556Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[2019-06-26T10:20:14.556Z] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[2019-06-26T10:20:14.556Z] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
[2019-06-26T10:20:14.556Z] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.Suite.runChild(Suite.java:128)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.Suite.runChild(Suite.java:27)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[2019-06-26T10:20:14.556Z] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
[2019-06-26T10:20:14.556Z] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

--- a/engine/src/test/java/io/zeebe/engine/processor/StreamProcessorTest.java
+++ b/engine/src/test/java/io/zeebe/engine/processor/StreamProcessorTest.java
@@ -42,6 +42,7 @@ import io.zeebe.protocol.record.RecordType;
 import io.zeebe.protocol.record.ValueType;
 import io.zeebe.protocol.record.intent.DeploymentIntent;
 import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.test.util.TestUtil;
 import io.zeebe.util.exception.RecoverableException;
 import io.zeebe.util.sched.ActorControl;
 import java.time.Duration;
@@ -485,26 +486,28 @@ public class StreamProcessorTest {
   public void shouldCreateSnapshot() throws Exception {
     // given
     final CountDownLatch processingLatch = new CountDownLatch(1);
-    streamProcessorRule.startTypedStreamProcessor(
-        (processors, state) ->
-            processors.onEvent(
-                ValueType.WORKFLOW_INSTANCE,
-                WorkflowInstanceIntent.ELEMENT_ACTIVATING,
-                new TypedRecordProcessor<UnifiedRecordValue>() {
-                  @Override
-                  public void processRecord(
-                      TypedRecord<UnifiedRecordValue> record,
-                      TypedResponseWriter responseWriter,
-                      TypedStreamWriter streamWriter,
-                      Consumer<SideEffectProducer> sideEffect) {
-                    processingLatch.countDown();
-                  }
-                }));
+    final StreamProcessor streamProcessor =
+        streamProcessorRule.startTypedStreamProcessor(
+            (processors, state) ->
+                processors.onEvent(
+                    ValueType.WORKFLOW_INSTANCE,
+                    WorkflowInstanceIntent.ELEMENT_ACTIVATING,
+                    new TypedRecordProcessor<UnifiedRecordValue>() {
+                      @Override
+                      public void processRecord(
+                          TypedRecord<UnifiedRecordValue> record,
+                          TypedResponseWriter responseWriter,
+                          TypedStreamWriter streamWriter,
+                          Consumer<SideEffectProducer> sideEffect) {
+                        processingLatch.countDown();
+                      }
+                    }));
 
     // when
     final long position =
         streamProcessorRule.writeWorkflowInstanceEvent(WorkflowInstanceIntent.ELEMENT_ACTIVATING);
     processingLatch.await();
+    TestUtil.waitUntil(() -> streamProcessor.getLastProcessedPositionAsync().join() > -1);
     streamProcessorRule.getClock().addTime(SNAPSHOT_INTERVAL);
 
     // then

